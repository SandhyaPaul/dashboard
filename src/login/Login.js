import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';

import { login } from '../redux/app-actions';

import './login.css';

class Login extends React.PureComponent {
	state = {
		userId: '',
		password: ''
	};

	componentWillReceiveProps(nextProps) {
		if (nextProps.loggedIn === true) {
			this.props.history.push('/app');
		}
	}

	updateValues = evt => {
		const types = {
			text: 'userId',
			password: 'password'
		};
		let toUpdate = types[evt.currentTarget.type]; //if text, then username, if type password,the psw
		this.setState({ [toUpdate]: evt.currentTarget.value });
	};

	doLogin = () => {
		let { userId, password } = this.state;
		this.props.login(userId, password);
	};

	render() {
		return (
			<div className="login-container-new">
				<div className="login-left-column">
					<img src="Dashboard.png" className="dashboard-image" />
					<div className="copyright-text">Â© Copyright 2018, News technolody rights reserved.</div>
				</div>
				<div className="login-right-column">
					<div className="login-container">
						<div className="login-form">
							{/*<div className="header-label">Login</div>*/}
							<label>Username</label>
							<div className="boxed-form">
								<input name="UserId" onChange={this.updateValues} />
							</div>
							<label>Password</label>
							<div className="boxed-form">
								<input name="Password" type="password" onChange={this.updateValues} />
							</div>
							{this.props.error ? <p className="login-error">Invalid credentials</p> : null}
							<div className="flex-display-normal">
								<button className="btn btn-block submit-btn" onClick={this.doLogin}>
									Sign In
								</button>
							</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export { Login as TestLogin };

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ login }, dispatch);
};

const mapStateToProps = state => {
	return {
		loggedIn: state.app.loggedIn,
		error: state.app.error
	};
};

export { mapDispatchToProps, mapStateToProps };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));
