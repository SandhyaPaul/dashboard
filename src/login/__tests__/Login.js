import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { TestLogin, mapDispatchToProps, mapStateToProps } from '../Login';

describe('DOM rendering', () => {
	it('renders LandingPage without breaking', () => {
		shallow(<TestLogin />);
	});

	it('DOM snapshot of LandingPage', () => {
		const wrapper = shallow(<TestLogin />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});

describe('Behavior checks', () => {
	it('redux actions', () => {
		const actions = mapDispatchToProps();
		expect(Object.keys(actions)).toEqual(['login']);
		expect(actions.login).toBeInstanceOf(Function);
	});
	it('gives the right props from state', () => {
		const expectedProps = {
			loggedIn: true
		};
		expect(mapStateToProps({ app: { loggedIn: true } })).toEqual(expectedProps);
	});
	it('Default state', () => {
		const wrapper = shallow(<TestLogin />);
		const expectedState = {
			userId: '',
			password: ''
		};
		expect(wrapper.state()).toEqual(expectedState);
	});
	it('onChange of userId input', () => {
		const wrapper = shallow(<TestLogin />);
		const mockEvt = {
			currentTarget: { type: 'text', value: 'test' }
		};
		wrapper.find('input[name="UserId"]').simulate('change', mockEvt);
		expect(wrapper.state('userId')).toEqual('test');
	});
	it('onChange of password input', () => {
		const wrapper = shallow(<TestLogin />);
		const mockEvt = {
			currentTarget: { type: 'password', value: '123' }
		};
		wrapper.find('input[type="password"]').simulate('change', mockEvt);
		expect(wrapper.state('password')).toEqual('123');
	});
	it('calls login function on click of submit button', () => {
		const fn = jest.fn();
		const wrapper = shallow(<TestLogin login={fn} />);
		wrapper.setState({ userId: 'test', password: '123' });
		wrapper.find('button').simulate('click');
		expect(fn.mock.calls.length).toEqual(1);
		expect(fn.mock.calls[0]).toEqual(['test', '123']);
	});
	it('Redirects to landing page, if after user logs in', () => {
		const historyMock = [];
		const wrapper = shallow(<TestLogin history={historyMock} />);
		wrapper.setProps({ loggedIn: true });
		expect(historyMock.length).toEqual(1);
		expect(historyMock[0]).toEqual('/app');
	});
	it('Doesnt redirect anywhere if user login fails', () => {
		const historyMock = [];
		const wrapper = shallow(<TestLogin history={historyMock} />);
		wrapper.setProps({ loggedIn: false });
		expect(historyMock).toEqual(historyMock);
	});
});
