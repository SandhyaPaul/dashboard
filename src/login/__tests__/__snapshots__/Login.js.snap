// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM rendering DOM snapshot of LandingPage 1`] = `
<div
  className="login-container"
>
  <div
    className="login-form"
  >
    <img
      alt="Company Logo"
      className="company-logo"
      src="Logo.png"
    />
    <div
      className="boxed-form"
    >
      <label>
        Username
      </label>
      <input
        name="UserId"
        onChange={[Function]}
      />
    </div>
    <div
      className="boxed-form"
    >
      <label>
        Password
      </label>
      <input
        name="Password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
